### PR TITLE
boards: stm32f3_disco: Use pins PA9/PA10 for I2C_2

### PR DESCRIPTION
**Pull request by ydamigos**
On stm32f3_disco pins PF0/PF1 are used for OSC_IN/OSC_OUT
signals. Use pins PA9/PA10 for I2C_2 port

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

### CODE REVIEW

**Comment by ydamigos** [comment]
@erwango Could you please have a look?

**Review by dbkinder — DISMISSED** [review]
+1 for doc change

File: drivers/pinmux/stm32/pinmux_stm32f3.h [hunk 1]
```diff
@@ -26,12 +26,12 @@
 #endif
 
 #if CONFIG_SOC_STM32F303XC
+#define STM32F3_PINMUX_FUNC_PA9_I2C2_SCL    (STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
+#define STM32F3_PINMUX_FUNC_PA10_I2C2_SDA   (STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
 #define STM32F3_PINMUX_FUNC_PB6_I2C1_SCL    (STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
 #define STM32F3_PINMUX_FUNC_PB7_I2C1_SDA    (STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
 #define STM32F3_PINMUX_FUNC_PB10_USART3_TX  (STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
 #define STM32F3_PINMUX_FUNC_PB11_USART3_RX  (STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
-#define STM32F3_PINMUX_FUNC_PF0_I2C2_SDA    (STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
```
Comment (erwango) [inline drivers/pinmux/stm32/pinmux_stm32f3.h]: Removing these defines is not required.
They are still valid and could be useful for other boards

**Review by erwango — DISMISSED** [review]

File: drivers/pinmux/stm32/pinmux_stm32f3.h (hunk 1 above)
Comment (ydamigos) [reply drivers/pinmux/stm32/pinmux_stm32f3.h]: Fixed

**Review by erwango — COMMENTED** [review]
Rebase required.
Otherwise LGTM

**Review by galak — APPROVED** [review]

**Pull request merged**